localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -878,6 +895,32 @@ Eigen::Matrix2d NDTScanMatcher::estimate_covariance(
   }
 }
 
+pcl::PointCloud<pcl::PointXYZRGB>::Ptr NDTScanMatcher::visualize_point_score(
+  const pcl::shared_ptr<pcl::PointCloud<PointSource>> & sensor_points_in_map_ptr,
+  const float & lower_nvs, const float & upper_nvs)
+{
+  pcl::PointCloud<pcl::PointXYZI> nvs_points_in_map_ptr_i;
+  nvs_points_in_map_ptr_i =
+    ndt_ptr_->calculateNearestVoxelScoreEachPoint(*sensor_points_in_map_ptr);
+  pcl::PointCloud<pcl::PointXYZRGB>::Ptr nvs_points_in_map_ptr_rgb{
+    new pcl::PointCloud<pcl::PointXYZRGB>};
```

Comment:
To avoid multiple memory allocations, I think it would be better to call reserve before the for loop.